Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,293 @@
+        internal void OnActivityStop(DiagnosticListener diagnosticListener)
+        {
+            Activity currentActivity = Activity.Current;
+            if (currentActivity == null)
+            {
+                DependencyCollectorEventSource.Log.CurrentActivityIsNull();
+                return;
+            }
+
+            DependencyCollectorEventSource.Log.TelemetryDiagnosticSourceListenerActivityStopped(currentActivity.Id, currentActivity.OperationName);
+
+            // extensibility point - can chain more telemetry extraction methods here
+            ITelemetry telemetry = this.ExtractDependencyTelemetry(currentActivity);
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            // properly fill dependency telemetry operation context
+            telemetry.Context.Operation.Id = currentActivity.RootId;
+            telemetry.Context.Operation.ParentId = currentActivity.ParentId;
+            telemetry.Timestamp = currentActivity.StartTimeUtc;
+
+            telemetry.Context.Properties["DiagnosticSource"] = diagnosticListener.Name;
+            telemetry.Context.Properties["Activity"] = currentActivity.OperationName.Substring(0, currentActivity.OperationName.Length - StopSuffix.Length);
+
+            this.client.Initialize(telemetry);
+
+            this.client.Track(telemetry);
+        }
+
+        internal DependencyTelemetry ExtractDependencyTelemetry(Activity currentActivity)
+        {
+            DependencyTelemetry telemetry = new DependencyTelemetry();
+
+            telemetry.Id = currentActivity.Id;
+            telemetry.Duration = currentActivity.Duration;
+
+            foreach (var item in currentActivity.Baggage)
+            {
+                if (!telemetry.Context.Properties.ContainsKey(item.Key))
+                {
+                    telemetry.Context.Properties[item.Key] = item.Value;
+                }
+            }
+
+            // get the Uri from event payload
+            Uri requestUri = null;
+            string dbStatement = null;
+            string httpMethodWithSpace = null;
+            string httpUrl = null;
+            string peerAddress = null;
+
+            foreach (KeyValuePair<string, string> tag in currentActivity.Tags)
+            {
+                // interpret Tags as defined by OpenTracing conventions
+                // https://github.com/opentracing/specification/blob/master/semantic_conventions.md
+                switch (tag.Key)
+                {
+                    case "operation.name": // not defined by OpenTracing
+                        {
+                            telemetry.Name = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "operation.data": // not defined by OpenTracing
+                        {
+                            telemetry.Data = tag.Value;
```

Comment:
is there any docs describing what Data property is for?
it seems to have some extended info about dependency (e.g. full url) or db statement. Is it correct?
Why we really need it in general case? What library should put there?
Why they cannot put whatever they want into the properties?

Reply:
they can put anything in tags, however Data is an elevated dependency property which should be specific to given technology - it can be "db.statement" or "http.url" for some services but for others it may be something entirely different (e.g. Redis, Event Hub)

currently there is no doc that specifies it but it will be described in the user guide for this

Reply:
https://docs.microsoft.com/en-us/azure/application-insights/application-insights-data-model-dependency-telemetry#data

Reply:
so, it seems the guidance for the lib owners should be: if you are http library, fill the http.url, if you are db, fill the db.statement. For the rest, use "operation.data".
Is it correct? It seems everyone could just use operation.data to simplify things. 
If it is so, I don't really see the value of supporting all possible opentracing tags. 
We can support them, but the guidance doc should probably tell about "operation.data".
Thoughts?

Reply:
That makes sense - in the guidance we will only mention **operation.data** but we will silently fall back to **http.url** / **db.statement**.